shop/models.py

```diff
@@ -0,0 +1,80 @@
+from datetime import timedelta
+from django.db import models, transaction
+from django.contrib.auth.models import AbstractUser
+from django.utils.timezone import now
+from snapShop import settings
+
+
+class User(AbstractUser):
+    wallet = models.DecimalField(
+        max_digits=10,
+        decimal_places=2,
+        default=settings.DEFAULT_USER_WALLET_BALANCE
+    )
+
+
+class Product(models.Model):
+    name = models.CharField(max_length=200)
+    description = models.TextField()
+    price = models.DecimalField(max_digits=10, decimal_places=2)
+    image = models.ImageField(upload_to="products/", blank=True, null=True)
+    stock = models.PositiveIntegerField()
+
+    def __str__(self):
+        return self.name
+
+    def purchase_product(self, user, quantity):
```

Comment:
It's better to separate interface into a dedicated function/method.
Current implementation has a weird behavior there `Product` instance modifies `User` instance directly. It's a bad practice from the code organization point of view, and violates SRP, OCP and DIP (SOLID principles).